### DISCUSSION TITLE
No link to PR is added when changelog platform is unsupported

### DISCUSSION THREAD

**Discussion by mueller-ma** [renovatebot/renovate#29398] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

> customChangelogUrl
> This doesn't help with fetching the changelogs, but if you configure it then Renovate will include a link to this URL in the PR body, so users can click through to read the changelog.

from https://docs.renovatebot.com/key-concepts/changelogs/#customchangelogurl

However this doesn't seem to work. I created a reproduction repo here: https://github.com/mueller-ma/renovate-reproduce

- Configure example.com as changelog: https://github.com/mueller-ma/renovate-reproduce/blob/main/renovate.json#L9
- PR without link to it: https://github.com/mueller-ma/renovate-reproduce/pull/5

Adding the link to the PR is quite useful in cases where the changelog isn't on a supported platform. It's much better for the user to click on link directly in the PR vs. having to go to a different place where the link can be added (e.g. Readme).

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Fetching changelog: https://example.com (11 -> 12) (branch="renovate/debian-12.x")
INFO: Unknown platform, skipping changelog fetching. (branch="renovate/debian-12.x")
{
  "sourceUrl": "https://example.com"
  "hostType": null
}

```

</details>

**Comment by rarkins** [comment]
I did a code review and noticed something strange. customChangelogUrl is used for fetching the changelog (it takes precedence over sourceUrl), but it's not massaged to `changelogUrl` for the purposes actually described in the docs. So essentially it does the opposite of the docs. I wonder if we should rename it and make it do what it says in the docs, or if anyone is relying on that behavior.

**Reply by mueller-ma** [reply, marked as answer]
- https://github.com/renovatebot/renovate/pull/29495
